source/conf.py

```diff
@@ -32,6 +32,8 @@
     "intrinsics": pathlib.Path(root, "data", "intrinsics.yml"),
 }
 
+sys.path.insert(0, str(root / "fortran_playground"))
```

Comment:
Could call the directory `extensions` instead (in case more are added at some point)?

Reply:
@henilp105 there is a bit of risk with this inheritance approach as well. But I see what you mean. I was thinking more of monkey-patching `CodeBlock` with a class that inherits from it (like you have done), and does your special `run` logic only if the language is `fortran`.

Reply:
@zmoon no, But we have restricted it with a different type of codeblock ```{play-code-block} , thus i have patched all the codeblocks in learn , except intrinsics and rosetta and if the code doesn't run ( in playground) then it would call the original codeblock 's run function, else our custom codeblocks with Exactly similar functionality would be run.